subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -0,0 +1,242 @@
+		}
+		memcpy(ssid->freq_list, channels_2_4ghz, ARRAY_SIZE(channels_2_4ghz) * sizeof(int));
+	}
+
+	if (creds->header.flags & WIFI_CREDENTIALS_FLAG_5GHz) {
+		ssid->freq_list = os_zalloc((ARRAY_SIZE(channels_5ghz) + 1) * sizeof(int));
+		if (!ssid->freq_list) {
+			LOG_ERR("unable to allocate buffer for freq_list");
+			return;
+		}
+		memcpy(ssid->freq_list, channels_5ghz, ARRAY_SIZE(channels_5ghz) * sizeof(int));
+	}
+}
+
+static inline int copy_psk(enum wifi_security_type security,
+			   const uint8_t *password,
+			   size_t password_len,
+			   struct wpa_ssid *ssid)
+{
+	if (!password) {
+		return 0;
+	}
+	switch (security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		ssid->key_mgmt = WPA_KEY_MGMT_NONE;
+		break;
+	case WIFI_SECURITY_TYPE_SAE:
+		ssid->key_mgmt = WPA_KEY_MGMT_SAE;
+		str_clear_free(ssid->sae_password);
+		ssid->sae_password = dup_binstr(password, password_len);
+
+		if (ssid->sae_password == NULL) {
+			LOG_ERR("%s:Failed to copy sae_password\n", __func__);
+			return -ENOMEM;
+		}
+		break;
+	case WIFI_SECURITY_TYPE_PSK:
+	case WIFI_SECURITY_TYPE_PSK_SHA256:
+		if (security == WIFI_SECURITY_TYPE_PSK) {
+			ssid->key_mgmt = WPA_KEY_MGMT_PSK;
+		} else {
+			ssid->key_mgmt = WPA_KEY_MGMT_PSK_SHA256;
+		}
+		str_clear_free(ssid->passphrase);
+		ssid->passphrase = dup_binstr(password, password_len);
+
+		if (ssid->passphrase == NULL) {
+			LOG_ERR("%s:Failed to copy passphrase\n", __func__);
+			return -ENOMEM;
+		}
+		break;
+	default:
+		LOG_ERR("%s:Security type %d is not supported\n", __func__, security);
+		return -EINVAL;
+	}
+	wpa_config_update_psk(ssid);
+
+	return 0;
+}
+
+static void add_stored_network(void *cb_arg, const char *ssid, size_t ssid_len)
+{
+	int ret = 0;
+	struct wpa_supplicant *wpa_s = cb_arg;
+	struct wpa_ssid *network = wpa_supplicant_add_network(wpa_s);
+	struct wifi_credentials_personal creds;
+
+	if (network == NULL) {
+		return;
+	}
+
+	network->ssid = os_zalloc(sizeof(u8) * WIFI_SSID_MAX_LEN);
+
+	if (network->ssid == NULL) {
+		goto exit;
+	}
+
+	/* load stored data */
+	ret = wifi_credentials_get_by_ssid_personal_struct(ssid, ssid_len, &creds);
+	if (ret) {
+		LOG_ERR("Loading WiFi credentials failed for SSID [%.*s], len: %d, err: %d",
+			ssid_len, ssid,  ssid_len, ret);
+		goto exit;
+	}
+
+	memcpy(network->ssid, creds.header.ssid, creds.header.ssid_len);
+	network->ssid_len = creds.header.ssid_len;
+	network->disabled = 1;
+	network->ieee80211w = 1;
+
+	/* apply priority */
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_FAVORITE) {
+		network->priority = 1;
+	}
+
+	/* apply BSSID restriction */
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_BSSID) {
+		memcpy(network->bssid, creds.header.bssid, WIFI_MAC_ADDR_LEN);
+		network->bssid_set = 1;
+	}
+	/* apply band restriction */
+	band_to_freq(&creds, network);
+
+	ret = copy_psk(creds.header.type, creds.password, creds.password_len, network);
+	if (ret) {
+		goto exit;
+	}
+
+	wpa_supplicant_enable_network(wpa_s, network);
+	return;
+exit:
```

Comment:
I'd rename this label to something more descriptive. There's a return above that seems to happen on success, while this is failure (?)

Reply:
renamed to failure